File: app/src/main/java/mx/dev/franco/automusictagfixer/common/ConnectionChecker.java
package mx.dev.franco.automusictagfixer.common;

import android.content.Context;
import android.os.AsyncTask;

import mx.dev.franco.automusictagfixer.interfaces.AsyncOperation;
import mx.dev.franco.automusictagfixer.utilities.AndroidUtils;

public class ConnectionChecker extends AsyncTask<Context, Void, Boolean> {
    private AsyncOperation<Void, Boolean, Void, Void> mCallback;

    public ConnectionChecker(AsyncOperation<Void, Boolean, Void, Void> callback) {
        mCallback = callback;
    }

    @Override
    protected Boolean doInBackground(Context... contexts) {
        return AndroidUtils.isConnected(contexts[0]);
    }

    @Override
    protected void onPostExecute(Boolean aBoolean) {
        if (mCallback != null)
            mCallback.onAsyncOperationFinished(aBoolean);
    }
}

File: app/src/main/java/mx/dev/franco/automusictagfixer/common/ConnectionChecker.kt
package mx.dev.franco.automusictagfixer.common

import android.content.Context
import android.os.AsyncTask
import mx.dev.franco.automusictagfixer.interfaces.AsyncOperation
import mx.dev.franco.automusictagfixer.utilities.AndroidUtils

class ConnectionChecker(private val mCallback: AsyncOperation<Void, Boolean, Void, Void>?) :
    AsyncTask<Context?, Void?, Boolean>() {
    override fun doInBackground(vararg contexts: Context?): Boolean {
        return AndroidUtils.isConnected(contexts[0])
    }

    override fun onPostExecute(aBoolean: Boolean) {
        mCallback?.onAsyncOperationFinished(aBoolean)
    }
}